### PR TITLE
Combine gold demo with final demo

### LINKED ISSUE #1670: Create Final Demo for presentation
Over the past Quarter, Custom-scripts has experimented with conditional WDL files, integrating Lua, and other ways of making Chiventure more dynamic. As we approach our final demonstration, this is the final demo that we will create and present during our final presentation. This issue should be closed when the final demo has been created that combines the smaller demos we have created over the course of this quarter.

### PR DESCRIPTION
**Pull request by RamarkoB**
This is a branch that combines the gold demo with the wizard/warrior demo. Once all three demos have been combined, this should be the final demo that we present during our final presentation

### CODE REVIEW

File: src/custom-scripts/examples/demo_2022.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    data_t temp = arg_t_get(togay);
+    custom_string = temp.s;
+
+    /* Where custom_type comes into play, create a dynamic string (hold different values) depending
+    on what the user enters at the start of the game */
+    data_t d1, d2;
+    data_t di1, di2;
+
+    int string_num1;
+    printf("You receive a premonition... how much gold do you see in your future? ");
```
Comment (jackjackkimkim) [inline src/custom-scripts/examples/demo_2022.c:75]: i love this integration of the gold demo!

**Review by jackjackkimkim — APPROVED** [review]
this looks great! now we can run our final demo with the gold/chest example

File: src/custom-scripts/examples/loadwdl2.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    char string_num;
+    printf("Enter either 1 or 2 (1 for wizard class, 2 for warrior class): ");
+    scanf("%c", &string_num); 
+    dc.c = string_num;
+    togay = obj_add_arg(togay,dc,CHAR_TYPE);
+    char* custom_string = (char*)malloc(100);
+    data_t temp = arg_t_get(togay);
+    custom_string = temp.s;
+    char wizard[] = "Wizard Class", warrior[] = "Warrior Class";
+    if(strcmp(wizard,custom_string)== 0){
```
Comment (ecoble) [inline src/custom-scripts/examples/loadwdl2.c:46]: Style spacing

**Review by ecoble — APPROVED** [review]

**Comment by RamarkoB** [comment]
We successfully combined the gold.c demo with the demo_2022.c file to create a larger demo. This is visible in the demo_2022.c file.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great job consolidating your demo code!

**Pull request merged**